modules/bigquery-dataset/main.tf

```diff
@@ -244,19 +244,19 @@ resource "google_bigquery_table" "default" {
   dynamic "time_partitioning" {
     for_each = try(each.value.partitioning.time, null) != null ? [""] : []
     content {
-      expiration_ms = each.value.partitioning.time.expiration_ms
-      field         = each.value.partitioning.field
-      type          = each.value.partitioning.time.type
+      expiration_ms            = each.value.partitioning.time.expiration_ms
+      field                    = each.value.partitioning.time.field
+      type                     = each.value.partitioning.time.type
+      require_partition_filter = each.value.partitioning.time.require_partition_filter
     }
   }
 }
 
 resource "google_bigquery_table" "views" {
-  depends_on          = [google_bigquery_table.default]
   for_each            = var.views
   project             = var.project_id
   dataset_id          = google_bigquery_dataset.default.dataset_id
-  table_id            = each.key
+  table_id            = google_bigquery_table.default[each.key].table_id
```

Comment:
Here should be ` google_bigquery_table.views[each.key].table_id`

Reply:
table_id should point to the table, what am I missing Wiktor?

Reply:
IMO the test should be updated setting the table _Countries_ instead of _population_
- here: https://github.com/devuonocar/cloud-foundation-fabric/blob/8fc510f0334f1c05a7137a1a025a2f603b5c76ea/tests/modules/bigquery_dataset/examples/views.yaml#L28C13-L28C13
- here: https://github.com/devuonocar/cloud-foundation-fabric/blob/8fc510f0334f1c05a7137a1a025a2f603b5c76ea/tests/modules/bigquery_dataset/examples/views.yaml#L30

Reply:
So my understanding of the intent was, that we are providing here `google_bigquery_table.default[each.key].table_id` to specify, that this view is using this table. But we do not specify which tables we use, other than query of the view itself.

Also, view can't have the same `table_id` as the table. They have to be unique.

Reply:
Taking again a look to the code I think you're right. I'm going to update it

Reply:
I think there is no good solution to avoid `depends_on` here. The only solution I can imagine, is to reference the resources while creating a query, which in this case is impossible. 

So I think in this case, we have to live with `depends_on` all tables.

Reply:
Thanks for being thorough Wiktor! :)

Reply:
Thanks both